storage/pebble.go

```diff
@@ -0,0 +1,136 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+
+	storageErrors "github.com/gnolang/tx-indexer/storage/errors"
+)
+
+var _ Storage = &Pebble{}
+
+// Storage is the instance of an embedded storage
+type Pebble struct {
+	db *pebble.DB
+}
+
+// New creates a new storage instance at the given path
+func NewPebble(path string) (*Pebble, error) {
+	db, err := pebble.Open(path, &pebble.Options{
+		// TODO: EventListener
```

Comment:
Leftover?

Just curious, what kind of options can we use for initializing pebble?

Reply:
You can tweak pebble ad-infinitum.

Some of the parameters that might be interesting to expose to the user/tweak for our use case are:
- Control the amount of data written to disk at the same time, using `BytesPerSync` and `WALBytesPerSync`
- Tweak the memory cache, by default is only 8MB
- EventListener will give us some information about events triggered by pebble under the hood, like compactions, flushes, and table deletions.
- We can provide our own FS implementation, allowing saving directly to S3 or just using memory.
- Tweak how big our persistence levels should be, allowing us to adapt better to different hardware.
- Inject our own logger and show pebble logs to the user
- and more...

Reply:
All of these seem amazing. I especially like the ability to let Pebble write directly to S3 🤯 

We can discuss in the future what we want to optimize more, I'm pleasantly surprised there is a treasure trove of options for it